examples/grpc-kotlin/src/main/kotlin/example/armeria/grpc/kotlin/HelloServiceImpl.kt

```diff
@@ -3,52 +3,50 @@ package example.armeria.grpc.kotlin
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.withContext
+
+/**
+ * Note that if you want to run [HelloServiceImpl] in an event loop and
+ * get a current [ServiceRequestContext],
+ * you should initialize [HelloServiceImpl] with [Dispatchers.Unconfined].
+ * See https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/-unconfined.html
+ * Unless [Dispatchers.Default] executor will handle this service.
+ */
+class HelloServiceImpl : HelloServiceGrpcKt.HelloServiceCoroutineImplBase(Dispatchers.Unconfined) {
 
     /**
      * Sends a [HelloReply] immediately when receiving a request.
      */
-    override fun hello(request: HelloRequest, responseObserver: StreamObserver<HelloReply>) {
-        responseObserver.onNext(buildReply(toMessage(request.name)))
-        responseObserver.onCompleted()
+    override suspend fun hello(request: HelloRequest): HelloReply {
+        // Make sure that current thread is request context aware
+        ServiceRequestContext.current();
+        return buildReply(toMessage(request.name))
     }
 
-    override fun lazyHello(request: HelloRequest, responseObserver: StreamObserver<HelloReply>) {
-        // You can use the event loop for scheduling a task.
-        ServiceRequestContext.current().contextAwareEventLoop().schedule({
-            responseObserver.onNext(buildReply(toMessage(request.name)))
-            responseObserver.onCompleted()
-        }, 3, TimeUnit.SECONDS)
+    override suspend fun lazyHello(request: HelloRequest): HelloReply {
+        delay(3000L)
+        return buildReply(toMessage(request.name))
     }
 
     /**
      * Sends a [HelloReply] using `blockingTaskExecutor`.
      *
      * @see [Blocking service implementation](https://line.github.io/armeria/server-grpc.html#blocking-service-implementation)
      */
-    override fun blockingHello(request: HelloRequest, responseObserver: StreamObserver<HelloReply>) {
-        // Unlike upstream gRPC-Java, Armeria does not run service logic in a separate thread pool by default.
-        // Therefore, this method will run in the event loop, which means that you can suffer the performance
-        // degradation if you call a blocking API in this method. In this case, you have the following options:
-        //
-        // 1. Call a blocking API in the blockingTaskExecutor provided by Armeria.
-        // 2. Set `GrpcServiceBuilder.useBlockingTaskExecutor(true)` when building your GrpcService.
-        // 3. Call a blocking API in the separate thread pool you manage.
-        //
-        // In this example, we chose the option 1:
-        ServiceRequestContext.current().blockingTaskExecutor().submit {
+    override suspend fun blockingHello(request: HelloRequest): HelloReply {
+        return withContext(Dispatchers.IO) {
```

Comment:
Not sure, but I guess it's more idiomatic for an armeria app to use the blocking task executor - it's context aware and is language-agnostic. Should we stick with `blockingTaskExecutor`, either using `submit` or `withContext(blockingTaskExecutor().asCoroutineDispatcher())`?

Reply:
I've updated before reading your review 🤣